core/node/core.go

```diff
@@ -41,18 +42,35 @@ func BlockService(lc fx.Lifecycle, bs blockstore.Blockstore, rem exchange.Interf
 // Pinning creates new pinner which tells GC which blocks should be kept
 func Pinning(bstore blockstore.Blockstore, ds format.DAGService, repo repo.Repo) (pin.Pinner, error) {
 	internalDag := merkledag.NewDAGService(blockservice.New(bstore, offline.Exchange(bstore)))
-	pinning, err := pin.LoadPinner(repo.Datastore(), ds, internalDag)
+	rootDS := repo.Datastore()
+
+	syncFn := func() error { return rootDS.Sync(blockstore.BlockPrefix) }
```

Comment:
No reason the pinner needs to sync the Filestore is there?

Reply:
Unfortunately, yes. When pinning, we may _reference_ blocks in the filestore.